cpp/unreal_plugins/SimulationController/Source/SimulationController/NavMesh.cpp

```diff
@@ -83,3 +90,72 @@ std::vector<uint8_t> NavMesh::getRandomPoints(int num_points)
     }
     return Std::reinterpretAs<uint8_t>(points);
 }
+
+std::vector<uint8_t> NavMesh::getReachablePoints(const std::vector<std::vector<float>>& start_points)
+{
+    std::vector<double> end_points;
+    for (auto& start_point : start_points) {
```

Comment:
Assert that `reference_points` is a multiple of 3, use an old-fashioned for-loop.

Reply:
Why would `reference_points` be a multiple of 3? User should be able to pass any number of `reference_points` to this function.
Do you mean the length of each vector in `reference_points` should be 3?

Reply:
I just saw your other comment. If we use `std::vector<double>` instead of `std::vector<std::vector<double>>`, we will need to assert the multiple of `3` condition.